StripePayments/StripePayments/Source/PaymentHandler/STPPaymentHandler.swift

```diff
@@ -1271,12 +1271,12 @@ public class STPPaymentHandler: NSObject {
             // The customer must authorize the transaction in their banking app within 1 minute
             // The merchant integration should spin and poll their backend or Stripe to determine success
             guard
-                let currentAction = self.currentAction
-                    as? STPPaymentHandlerPaymentIntentActionParams
+                let presentingVC = currentAction.authenticationContext
+                    as? PaymentSheetAuthenticationContext
             else {
-                fatalError()
+                return
             }
-            currentAction.complete(with: .succeeded, error: nil)
+            presentingVC.presentPollingVCForAction(action: currentAction, type: .blik)
 
```

Comment:
This is a little tricky! If we are using PaymentSheet we want to use the PollingViewController, otherwise we should use the existing logic for API bindings users. I would recommend doing something like by checking if we have a payment sheet context like we do for UPI but add the existing logic into the body of the guard

```swift
guard
    let presentingVC = currentAction.authenticationContext
        as? PaymentSheetAuthenticationContext
else {
   // Do existing bindings logic
}

presentingVC.presentPollingVCForAction(action: currentAction, type: .blik)
```


Read the last paragraph in this docs section for context: https://stripe.com/docs/payments/blik/accept-a-payment?platform=mobile&ui=ios#ios-submit-payment

Add some comments here so future devs know why we're doing this.